src/Raven.Server/Documents/Indexes/Debugging/IndexDebugExtensions.cs

```diff
@@ -435,6 +435,8 @@ public static string[] GetEntriesFields(this Index self)
                     return ((AutoMapIndex)self).Definition.IndexFields.Keys.ToArray();
                 case IndexType.AutoMapReduce:
                     return ((AutoMapReduceIndex)self).Definition.IndexFields.Keys.ToArray();
+                case IndexType.Faulty:
+                    throw new ArgumentException($"Index '{self.Name}' is Faulty. Can't return index entries for a faulty index.");
```

Comment:
InvalidOperatonException would be better here

Reply:
done